core/src/main/java/com/expedia/www/haystack/client/idgenerators/IdGenerator.java

```diff
@@ -17,5 +17,7 @@
 package com.expedia.www.haystack.client.idgenerators;
 
 public interface IdGenerator {
-    Object generate();
+    Object generateTraceId();
```

Comment:
To remain backward compatible, can you keep the generate function in the interface?

And add the other two methods generateTraceId and generateSpanId with the default implementation, calling   `generate()`.

Also, If you do that,  LongIdGenerator changes in this PR may not be required.

Reply:
thanks, will do that.

Reply:
it's a bit confusing now after the changes :)

even though they don't need to, the generator implementations _can_ override 3 methods: generate, generateSpanId and generateTraceId.
I suggest annotating the generate method with deprecated so that we can remove it in the future.

Reply:
Yes. We can mark it deprecated.